### ISSUE TITLE
Restsharp adds UserAgent headers to HttpClient instance that is passed explicitly

### ISSUE THREAD

**Issue by p-skakun** [restsharp/RestSharp#1990]
**Describe the bug**
When create multiple `RestClient` instances with the same HttpClient instance and default options it adds `UserAgent` header each time.
In certain scenarios it causes `System.InvalidOperationException: Collection was modified`

**To Reproduce**
```
var httpClient = new HttpClient();
var restClient1 = new RestClient(httpClient);
var restClient2 = new RestClient(httpClient);
Assert.AreEqual(2, httpClient.DefaultRequestHeaders.UserAgent.Count);
Assert.AreEqual(httpClient.DefaultRequestHeaders.UserAgent.First(), httpClient.DefaultRequestHeaders.UserAgent.Last());
```

**Expected behavior**
Reused httpClient keeps unique `UserAgent` headers

**Desktop (please complete the following information):**
 - OS: Windows 11 Home
 - .NET version: 7
 - Version: 109.0.0-preview.1

**Additional context**
I believe the issues starts here: 
https://github.com/restsharp/RestSharp/blob/dev/src/RestSharp/RestClient.cs#L131

In case of default `UserAgent` value `Product?.Name` is `RestSharp`, while `UserAgent` is `RestSharp/x.x.x` so the check always passes.

**Comment by tacosontitan** [comment]
`ConfigureHttpClient` is not called in the constructor utilized by your reproduction steps. However, this is reproducible:

```csharp
// arrange
var httpClient = new HttpClient();
var clientOptions = new RestClientOptions();

// act
var restClient1 = new RestClient(httpClient, clientOptions);
var restClient2 = new RestClient(httpClient, clientOptions);

// assert
Assert.Contains(httpClient.DefaultRequestHeaders.UserAgent, agent => $"{agent.Product.Name}/{agent.Product.Version}" == clientOptions.UserAgent);
Assert.Single(httpClient.DefaultRequestHeaders.UserAgent);
```
In `RestClient`'s current state, this test *will* fail. This is caused by the fact that the current snippet only compares the `Product` property with the `options.UserAgent` which also contains a version.

https://github.com/restsharp/RestSharp/blob/f7d3984244c4808ae7c068e028753c2857f325b9/src/RestSharp/RestClient.cs#L128-L136

Note that line 131 here is the source of the issue.

**Comment by alexeyzimarev** [comment]
Fixed by @tacosontitan in #2008, I released it in 109.0.1.
